aea/connections/stub/connection.py

```diff
@@ -253,7 +253,7 @@ async def disconnect(self) -> None:
 
         assert self.in_queue is not None, "Input queue not initialized."
         self._observer.stop()
-        self._observer.join()
+        self._observer.join()  # TODO: blocking code?
```

Comment:
@solarw please open an issue and remove the comment, otherwise the hash will be invalidated

Reply:
done